model/src/main/kotlin/utils/FileArchiver.kt

```diff
@@ -30,6 +30,7 @@ import org.ossreviewtoolkit.model.KnownProvenance
 import org.ossreviewtoolkit.utils.common.FileMatcher
 import org.ossreviewtoolkit.utils.common.collectMessages
 import org.ossreviewtoolkit.utils.common.packZip
+import org.ossreviewtoolkit.utils.common.safeDeleteRecursively
```

Comment:
commit: The operating system takes care of deleting temp files anyway.
So, the only reason I can see for deleting files manually is freeing up disk space.
In that regard deleting all the created temp file manually makes sense to me.

For empty directories, I doubt that we should add mechanims for their deletion as these will be deleted anyway? What do you think?

Reply:
> The operating system takes care of deleting temp files anyway.

Well, that depends on the OS and configuration. Usually temporary files are not deleted immediately, but after some hours / days, or only on the next reboot.

> So, the only reason I can see for deleting files manually is freeing up disk space.

I see people being confused by `ort-*` directory left-overs in the OS's temporary directory because that creates the impression that something went wrong and ORT crashed.

So it's also a way to double-check that nothing unexpected did happen.

Reply:
I see. So let's leave it as-is.

Anyhow, what do you think of doing something similar as `File.deleteOnExit()` inside `createOrtTempDir()`. E.g. register a hook triggered when JVM exits, which then deletes all still existing temp dirs recursively?

Reply:
Regarding the OS deleting the temp files automagically: it really seems to depend on the system. I'm running ORT and developing plugins for it on WSL2, and it surely isn't very eager to delete anything from the /tmp/.  It's gone to a stage where I actually had to create a Bash script to controllably delete ORT-specific directories from it.

Reply:
> doing something similar as `File.deleteOnExit()`

I'd rather not, as [`java.io.File.deleteOnExit()` can cause leaks](http://fxapps.blogspot.com/2017/10/how-javaiofiledeleteonexit-can-cause.html).